### PR TITLE
browser sdks re-structure

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

Restructures Browser SDK section/ renames TypeScript Browser SDK.

### CODE REVIEW

**Review by caseyamp — APPROVED** [review]

File: docs/data/sdks/javascript/index.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ icon: simple/javascript
 This is the official documentation for the Amplitude Analytics JavaScript SDK.
 
 !!!deprecated "Legacy SDK"
-    This is a legacy SDK and will only receive bug fixes until deprecation. Upgrade to the [TypeScript Browser SDK](../typescript-browser/) which supports plugins and more. See the [Migration Guide](../../sdks/typescript-browser/migration) for more help.
+    This is a legacy SDK and will only receive bug fixes until deprecation. Upgrade to the [Browser SDK](../typescript-browser/) which supports plugins and more. See the [Migration Guide](../../sdks/typescript-browser/migration) for more help.
```
Comment (justin-fiedler) [inline docs/data/sdks/javascript/index.md:13]: I think adding "latest" reads better.

```suggestion
    This is a legacy SDK and will only receive bug fixes until deprecation. Upgrade to the latest [Browser SDK](../typescript-browser/) which supports plugins and more. See the [Migration Guide](../../sdks/typescript-browser/migration) for more help.
```

File: docs/data/sdks/marketing-analytics-browser/index.md [hunk 2]
```diff
@@ -13,9 +13,9 @@ The Marketing Analytics Browser SDK extends the Browser SDK to identify users an
 
     [:material-github: GitHub](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/marketing-analytics-browser) · [:material-code-tags-check: Releases](https://github.com/amplitude/Amplitude-TypeScript/releases?q=marketing-analytics-browser&expanded=true) · [:material-book: API Reference](https://amplitude.github.io/Amplitude-TypeScript/modules/_amplitude_marketing_analytics_browser.html)
 
-!!!note "Marketing Analytics Browser versus the Typescript Broswer SDK"
+!!!note "Marketing Analytics Browser versus the Broswer SDK"
```
Comment (justin-fiedler) [inline docs/data/sdks/marketing-analytics-browser/index.md:16]: Suggestion + typo
```suggestion
!!!note "Marketing Analytics SDK versus the Browser SDK"
```

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
LGTM, except the typo. Left some suggestions on wording.

File: docs/data/sdks/javascript/index.md (hunk 1 above)
Comment (yuhao900914) [reply docs/data/sdks/javascript/index.md:13]: Agree.

File: docs/data/sdks/marketing-analytics-browser/index.md (hunk 2 above)
Comment (yuhao900914) [reply docs/data/sdks/marketing-analytics-browser/index.md:16]: Probably, it should be  `Marketing Analytics Browser SDK`?

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**